llvm/test/CodeGen/AMDGPU/sgpr-spill-overlap-wwm-reserve.mir

```diff
@@ -1,5 +1,5 @@
 # NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
-# RUN: llc -mtriple=amdgcn-unknown-amdpal -mcpu=gfx1030 -start-before=si-lower-sgpr-spills -stop-after=prologepilog -verify-machineinstrs --stress-regalloc=2 -o - %s | FileCheck -check-prefix GCN %s
+# RUN: llc -mtriple=amdgcn-unknown-amdpal -mcpu=gfx1030 -start-before=si-lower-sgpr-spills -stop-after=prologepilog -verify-machineinstrs --stress-regalloc=5 -o - %s | FileCheck -check-prefix GCN %s
```

Comment:
To explain this test change:
--stress-regalloc=2 limits the VGPR class to 2 registers *after* reserve registers are subtracted.
Without reservation by SILowerSGPRSpills the 3 VGPRs count against the VGPR class usage, so the class size must be enlarged (to 5) to accommodate them.

Reply:
To preempt an obvious follow up question: "how does SILowerSGPRSpill exceed the register limit?"
`allocateSGPRSpillToVGPRLane` uses `findUnusedRegister` which does not consult the allocation in `RegisterClassInfo` which imposes the limit.

Reply:
This option should have been better integrated into MRI to fully limit the register usage in various other passes.